packages/functional_widget/lib/src/parameters.dart

```diff
@@ -4,10 +4,24 @@ import 'package:analyzer/dart/element/nullability_suffix.dart';
 import 'package:analyzer/dart/element/type.dart' as element_type;
 import 'package:build/build.dart';
 import 'package:code_builder/code_builder.dart';
+import 'package:collection/collection.dart';
 
 class FunctionParameters {
   FunctionParameters._(this._parameters);
 
+  static const nonUserDefinedTypeSymbols = [
+    'Key',
+    'Key?',
```

Comment:
What is the purpose of allowing non-nullable key if later we do `key!` to pass it?

Reply:
Ignore me, I didn't see that you added the `required Key key`